### PR TITLE
[125] Add functionality for additional funding variables

### LINKED ISSUE #125: Add additional funding variables to investments predictions dataset creation
Add variables for:
* raised_amount_total_usd
* last_round_investment_type
* last_round_raised_amount_usd
* number_of_months_before_first_investment (operating speed)
* last_round_post_money_valuation – only available for about 5% of funding rounds. Still worth including?

### PR DESCRIPTION
**Pull request by Jack-0-0**
Closes #125.

This PR adds into the investment prediction dataset variables for:
* `total_investment_amount_usd`
* `last_investment_round_usd`
* `last_investment_round_type`
* `n_months_before_first_investment`

### CODE REVIEW

**Comment by Jack-0-0** [comment]
I could also easily add in `last_round_post_money_valuation` but it is only available for about 5% of the records, do you think it is worth adding?

**Comment by beingkk** [comment]
Hi, I will aim to review by end of Monday. Hope that works!

File: innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py [hunk 1]
```diff
@@ -169,14 +177,49 @@ def create_dataset(
         # Create future_success variable which is set to 1 if one of the above flags is 1
         .assign(future_success=lambda x: x[SUCCESS_COLS].max(axis=1))
         .drop(columns=SUCCESS_COLS)
+        # Add col for lastest funding date in window
```
Comment (beingkk) [inline innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py]:
```suggestion
        # Add col for latest funding date in window
```

File: innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py [hunk 2]
```diff
@@ -169,14 +177,49 @@ def create_dataset(
...
+        # Add col for first funding date in window
+        .pipe(
+            utils.add_first_last_date,
+            "funding_round_date",
+            False,
+            window_start_date,
+            window_end_date,
+            "first_funding_date_in_window",
+        )
+        # Add col for last funding round in window
```
Comment (beingkk) [inline innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py]: What information does this column hold? (funding amount, date, ...?)

**Review by beingkk — APPROVED** [review]
Hey, thanks for this! I successfully ran it on my machine, and the code looks good to me.

Responding to your comment about the extra feature: I think adding the valuation variable would be interesting but we can park it for now.

Perhaps one small thing: It might be useful to add usage instructions at the top of the `create_dataset.py` file, similar as in the previous PR. For example, showing how to specify the parameters, and noting the expected run time (I think it was about 15 minutes on my machine).

File: innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py (hunk 1 above)
Comment (Jack-0-0) [reply innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py]: Good spot, thanks!

File: innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py (hunk 2 above)
Comment (Jack-0-0) [reply innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset/create_dataset.py]: This is for funding round number. I have updated the comment.

**Comment by Jack-0-0** [comment]
> Hey, thanks for this! I successfully ran it on my machine, and the code looks good to me.
> 
> Responding to your comment about the extra feature: I think adding the valuation variable would be interesting but we can park it for now.
> 
> Perhaps one small thing: It might be useful to add usage instructions at the top of the `create_dataset.py` file, similar as in the previous PR. For example, showing how to specify the parameters, and noting the expected run time (I think it was about 15 minutes on my machine).

Thank you for reviewing @beingkk. I have updated the comments and the docstring at the top of the file as suggested 🙂

**Pull request merged**